src/controller/cve.controller/cve.controller.js

```diff
@@ -67,35 +67,48 @@ async function submitCve (req, res, next) {
 }
 
 // Updates a CVE if one exists for the specified CVE ID
-async function updateCve (req, res) {
-  const cve = req.body
-  const cveId = req.params.id
+async function updateCve (req, res, next) {
+  try {
+    const cve = req.ctx.body
+    const cveId = req.ctx.params.id
+    const cveRepo = req.ctx.repositories.getCveRepository()
+    const orgRepo = req.ctx.repositories.getOrgRepository()
 
-  if (cveId !== cve.CVE_data_meta.ID) {
-    return res.status(400).json({ message: cveId + ' parameter does not match ' + cve.CVE_data_meta.ID + ' in the JSON body.' })
-  }
+    // cve id in body must match id in URL param
+    if (cveId !== cve.CVE_data_meta.ID) {
+      return res.status(400).json(error.cveIdMismatch())
+    }
+
+    // update CVE record if it exists
+    const newCve = new Cve({ cve: cve })
+    const result = await cveRepo.updateByCveId(cveId, newCve)
+
+    // check if CVE record was found and updated
+    if (result.n === 0) {
+      logger.info(cveId + ' does not exist.')
+      return res.status(403).json(error.cveDne())
```

Comment:
This should return another error indicating that the CVE record does not exist (cve record != cve id). The `cveDNE` error indicates that a CVE ID does not exist, which is appropriate for creating a CVE record but should not be used when updating a CVE record.